src/exports.cc

```diff
@@ -872,6 +885,351 @@ Word wasi_unstable_clock_time_get(Word clock_id, uint64_t /*precision*/,
+    return 8; // __WASI_ERRNO_BADF - Bad file descriptor
+  }
+  
+  // For stdout and stderr, there's no need to sync as they're handled by the host system
+  return 0; // __WASI_ESUCCESS
+}
+
+// __wasi_errno_t __wasi_fd_tell(__wasi_fd_t fd, __wasi_filesize_t *retptr0);
+Word wasi_unstable_fd_tell(Word fd, Word retptr0) {
+  auto *context = contextOrEffectiveContext();
+  
+  // fd_tell is used to get the current offset of a file descriptor.
+  // Since we don't have a real file system in proxy-wasm, we can just return an error.
+  
+  // We only support stdout and stderr in proxy-wasm, which don't support seeking
+  if (fd != 1 /* stdout */ && fd != 2 /* stderr */) {
+    return 8; // __WASI_ERRNO_BADF - Bad file descriptor
+  }
+  
+  // For stdout and stderr, we'll just return 0 as the offset
+  if (!context->wasm()->setDatatype(retptr0, uint64_t(0))) {
+    return 21; // __WASI_EFAULT
+  }
+  
+  return 0; // __WASI_ESUCCESS
+}
+
+// __wasi_errno_t __wasi_path_create_directory(__wasi_fd_t fd, const char *path);
+Word wasi_unstable_path_create_directory(Word fd, Word path_ptr, Word path_len) {
+  // path_create_directory is used to create a directory.
+  // Since we don't have a real file system in proxy-wasm, we can just return an error.
+  
+  return 58; // __WASI_ENOTSUP - Not supported
+}
+
+// __wasi_errno_t __wasi_path_filestat_set_times(__wasi_fd_t fd, __wasi_lookupflags_t flags, const char *path, __wasi_timestamp_t atim, __wasi_timestamp_t mtim, __wasi_fstflags_t fst_flags);
+Word wasi_unstable_path_filestat_set_times(Word fd, Word flags, Word path_ptr, Word path_len, uint64_t atim, uint64_t mtim, Word fst_flags) {
+  // path_filestat_set_times is used to set the access and modification times of a file by path.
+  // Since we don't have a real file system in proxy-wasm, we can just return an error.
+  
+  return 58; // __WASI_ENOTSUP - Not supported
+}
+
+// __wasi_errno_t __wasi_path_link(__wasi_fd_t old_fd, __wasi_lookupflags_t old_flags, const char *old_path, __wasi_fd_t new_fd, const char *new_path);
+Word wasi_unstable_path_link(Word old_fd, Word old_flags, Word old_path_ptr, Word old_path_len, Word new_fd, Word new_path_ptr) {
+  // path_link is used to create a hard link.
+  // Since we don't have a real file system in proxy-wasm, we can just return an error.
+  
+  return 58; // __WASI_ENOTSUP - Not supported
+}
+
+// __wasi_errno_t __wasi_path_readlink(__wasi_fd_t fd, const char *path, uint8_t *buf, __wasi_size_t buf_len, __wasi_size_t *retptr0);
+Word wasi_unstable_path_readlink(Word fd, Word path_ptr, Word path_len, Word buf_ptr, Word buf_len, Word retptr0) {
+  auto *context = contextOrEffectiveContext();
+  
+  // path_readlink is used to read the contents of a symbolic link.
+  // Since we don't have a real file system in proxy-wasm, we can just return an error.
+  
+  // Set the number of bytes read to 0
+  if (!context->wasmVm()->setWord(retptr0, Word(0))) {
+    return 21; // __WASI_EFAULT
+  }
```

Comment:
The general convention is that return values aren't expected to be legitimate if the function returns error, so this shouldn't be needed.